crypto/src/lib.rs

```diff
@@ -697,16 +728,9 @@ impl<'de> Deserialize<'de> for PrivateKey {
     {
         use serde::de::Error as _;
 
-        #[derive(Deserialize)]
-        struct PrivateKeyCandidate {
-            algorithm: Algorithm,
-            payload: String,
-        }
-
-        // NOTE: Verify that private key is valid
-        let private_key = PrivateKeyCandidate::deserialize(deserializer)?;
-        Self::from_hex(private_key.algorithm, private_key.payload.as_ref())
-            .map_err(D::Error::custom)
+        let PrivateKeySerialized { algorithm, payload } =
+            PrivateKeySerialized::deserialize(deserializer)?;
+        Self::from_raw(algorithm, &payload).map_err(D::Error::custom)
```

Comment:
check out `serde(try_from` in `data_model/src/account.rs`. Is that applicable in this case?

Reply:
Thanks for pointing this out. `try_from` and `into` work perfectly here.